game/level_switch.gd

```diff
@@ -0,0 +1,21 @@
+extends Node2D
+
+var level_path: String setget _set_level_path
+
+# Called when the node enters the scene tree for the first time.
+func _ready():
+	rset_config('level_path', MultiplayerAPI.RPC_MODE_REMOTESYNC)
+
+func set_level(new_path: String):
+	rset('level_path', new_path)
+
+func _set_level_path(new_path: String):
+	level_path = new_path
+	for child in get_children():
+		# take care to not delete the sync component
+		if child.name != 'sync':
+			self.remove_child(child)
+			child.queue_free()
```

Comment:
would it be enough to remember the `name` of the new_level after once it's been added and then only remove that one node? (changes nothing in practice, but might make this more explicit)

Reply:
I think that players are also children of `self` in this context, so only deleting the level would not remove+respawn the players.

That behavior might be desired in some cases, but this way a level switch is rather clean.

Reply:
Ah good point, let's leave it like this for the moment then :)